server/build.gradle

```diff
@@ -212,6 +216,8 @@ thirdPartyAudit.ignoreMissingClasses(
   // from log4j
   'com.conversantmedia.util.concurrent.DisruptorBlockingQueue',
   'com.conversantmedia.util.concurrent.SpinPolicy',
+  //TODO PG why it fails now with ecs??
```

Comment:
Do we need to do anything about this TODO?

Reply:
the https://github.com/elastic/ecs-logging-java/blob/master/log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/EcsLayout.java#L90 is calling https://github.com/elastic/ecs-logging-java/blob/master/log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/ObjectMessageJacksonSerializer.java#L51
which in turn uses the `SerializationFeature`
 https://github.com/elastic/ecs-logging-java/blob/master/log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/ObjectMessageJacksonSerializer.java#L73

We don't depend on jackson.databind. As far as I remember it is because of the security requirements it has.

Because of this, when starting up we throw a `ClassNotFoundException` 
```
java.lang.ClassNotFoundException: com.fasterxml.jackson.databind.ObjectMapper
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:604)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:333)
	at co.elastic.logging.log4j2.ObjectMessageJacksonSerializer$Resolver.resolve(ObjectMessageJacksonSerializer.java:47)
	at co.elastic.logging.log4j2.EcsLayout.<init>(EcsLayout.java:90)
	at co.elastic.logging.log4j2.EcsLayout.<init>(EcsLayout.java:63)
	at co.elastic.logging.log4j2.EcsLayout$Builder.build(EcsLayout.java:408)
	at org.elasticsearch.common.logging.ECSJsonLayout$Builder.build(ECSJsonLayout.java:63)
	at org.elasticsearch.common.logging.ECSJsonLayout$Builder.build(ECSJsonLayout.java:45)
	at org.apache.logging.log4j.core.config.plugins.util.PluginBuilder.build(PluginBuilder.java:123)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createPluginObject(AbstractConfiguration.java:959)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:899)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:891)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.createConfiguration(AbstractConfiguration.java:891)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.doConfigure(AbstractConfiguration.java:514)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.initialize(AbstractConfiguration.java:238)
	at org.apache.logging.log4j.core.config.AbstractConfiguration.start(AbstractConfiguration.java:250)
	at org.apache.logging.log4j.core.LoggerContext.setConfiguration(LoggerContext.java:547)
	at org.apache.logging.log4j.core.LoggerContext.start(LoggerContext.java:263)
	at org.elasticsearch.common.logging.LogConfigurator.configure(LogConfigurator.java:236)
	at org.elasticsearch.common.logging.LogConfigurator.configure(LogConfigurator.java:129)
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:355)
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:170)
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:161)
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:86)
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:127)
	at org.elasticsearch.cli.Command.main(Command.java:90)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:126)
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:92)
```
This is then handled by the `ObjectMessageJacksonSerializer.Resolver` which returns null. EcsLayout works ok with that field being null (guarded in other usages).

I guess it is safe to just ignore `SerializationFeature` not being present. We already ignore `  'com.fasterxml.jackson.databind.ObjectMapper` which is used (and fails) first.

cc @felixbarny

Reply:
I don't really understand what this does. Is this a feature in the Elasticsearch build that checks dependencies for missing transitive dependencies?

It's safe to ignore `com.fasterxml.jackson.databind.SerializationFeature` as it's an optional dependency. If it's not available at runtime, it will be gracefully handled by the ecs logger.

Reply:
@felixbarny 
> Is this a feature in the Elasticsearch build that checks dependencies for missing transitive dependencies?
yes. we explicitly declare all transient dependencies in elasticsearch. At situations like this one, dependency cannot be included, but it is not being used at runtime.

thank you for confirming that it is safe with `SerializationFeature`